### PR TITLE
[FIRRTL][Emitter] Sync to use "latest", and bump that to 3.1.0.

### PR DESCRIPTION
**Pull request by dtzSiFive**
Match behavior of `mlirExportFIRRTL`.

### CODE REVIEW

File: include/circt/Dialect/FIRRTL/FIRParser.h [hunk 1]
```diff
@@ -104,7 +104,7 @@ struct FIRVersion {
 
   static FIRVersion defaultFIRVersion() { return {1, 0, 0}; }
 
-  static FIRVersion latestFIRVersion() { return {3, 0, 0}; }
+  static FIRVersion latestFIRVersion() { return {3, 1, 0}; }
```
Comment (seldridge) [inline include/circt/Dialect/FIRRTL/FIRParser.h:107]: Thanks!

File: lib/Dialect/FIRRTL/Export/FIREmitter.cpp [hunk 2]
```diff
@@ -1344,7 +1344,8 @@ void circt::firrtl::registerToFIRFileTranslation() {
   static mlir::TranslateFromMLIRRegistration toFIR(
       "export-firrtl", "emit FIRRTL dialect operations to .fir output",
       [](ModuleOp module, llvm::raw_ostream &os) {
-        return exportFIRFile(module, os, targetLineLength, {3, 1, 0});
+        return exportFIRFile(module, os, targetLineLength,
+                             FIRVersion::latestFIRVersion());
```
Comment (seldridge) [inline lib/Dialect/FIRRTL/Export/FIREmitter.cpp:1348]: Nice!

**Review by seldridge — APPROVED** [review]
LGTM

**Comment by dtzSiFive** [comment]
Thanks for (quick!) review! :tada:

**Pull request merged**